src/shared/components/ui/date-picker/DatePickerDate.tsx

```diff
@@ -0,0 +1,152 @@
+import { cn } from '@/shared/libs/cn';
+
+import { useDatePickerContext } from './DatePickerContext';
+
+interface DatePickerDateProps {
+  /**
+   * 날짜 버튼들이 배치되는 grid 컨테이너의 커스텀 클래스
+   */
+  gridClassName?: string;
+
+  /**
+   * 개별 날짜 버튼에 적용할 커스텀 클래스
+   */
+  dateButtonClassName?: string;
+
+  /**
+   * 예약 가능한 날짜 리스트
+   */
+  reservableDates?: string[];
+}
+
+/**
+ * DatePickerDate 컴포넌트
+ * 날짜들을 렌더링하고 선택 가능한 DatePicker의 날짜 그리드 컴포넌트입니다.
+ *
+ * @param {string} [props.gridClassName] - 날짜들이 배치되는 grid 컨테이너에 적용할 추가 CSS 클래스입니다.
+ * @param {string} [props.dateButtonClassName] - 개별 날짜 버튼에 적용할 추가 CSS 클래스입니다.
+ * @param {string[]} [props.reservableDates] - 예약 가능한 날짜들의 문자열 배열입니다. 해당 prop은 선택 옵션이며 선택하지 않으면, 현재를 기준으로 미래의 날짜들은 확인 가능합니다. (예: ['2025-07-20'])
+ *
+ * @example
+ * ```tsx
+ * <DatePicker.Date reservableDates={['2025-07-20', '2025-07-21']} />
+ * ```
+ *
+ * @returns {JSX.Element} 날짜 버튼들로 구성된 grid 레이아웃을 반환합니다.
+ */
+export default function DatePickerDate({
+  gridClassName,
+  dateButtonClassName,
+  reservableDates,
+}: DatePickerDateProps) {
+  const { currentMonth, today, onDateClick, selectedDate, size } =
+    useDatePickerContext();
+
+  // 현재 월의 시작일과 종료일 계산
+  const startOfMonth = currentMonth.startOf('month');
+  const endOfMonth = currentMonth.endOf('month');
+
+  // 달력은 항상 "주 단위(7일)"로 표현되므로,
+  // 월의 시작일 기준으로 그 주의 '일요일'까지 확장하고,
+  // 월의 종료일 기준으로 그 주의 '토요일'까지 확장합니다.
+  const startDate = startOfMonth.startOf('week');
+  const endDate = endOfMonth.endOf('week');
+
+  // startDate부터 endDate까지 하루씩 증가하며 날짜 목록 생성
+  const dateList = [];
+  let date = startDate;
+  while (date.isBefore(endDate) || date.isSame(endDate, 'day')) {
+    dateList.push(date);
+    date = date.add(1, 'day');
+  }
```

Comment:
`dateList` 배열과 그 안의 날짜 상태(`isToday`, `isSelected` 등)가 컴포넌트가 리렌더링될 때마다 처음부터 다시 계산됩니다.
`dateList` 배열 생성을 `useMemo`로 감싸 `currentMonth`가 바뀔 때만 재계산하도록 해서 최적화할 수 있습니다.

```suggestion
  const dateList = useMemo(() => {
    const startOfMonth = currentMonth.startOf('month');
       ... (기존 로직)
    return list;
  }, [currentMonth]);
```